### ISSUE TITLE
Target net standard

### ISSUE THREAD

**Issue by jjxtra** [EasyHook/EasyHook#278]
Support for .net standard 2.0 would be great. This would allow .net framework compatibility along with .net core.

**Comment by NotoriousRebel** [comment]
Any updates on this?

**Comment by spazzarama** [comment]
@NotoriousRebel I will probably be looking at this over the next month

**Comment by spazzarama** [comment]
It might be necessary to target .NET Core / .NET Framework, e.g. for EasyLoad AppDomain creation

**Comment by RussKie** [comment]
Joining the queue here. 

[Git Extensions](https://github.com/gitextensions/gitextensions) project has taken a dependency on EasyHook to enable dark theming of the app. 
Now I'm migrating the app the .NET Core 3.1 (https://github.com/gitextensions/gitextensions/pull/7162), and finding that EasyHook on remoting, that has been deprecated and removed in .NET Core.

So it would be great to have a .NET Core compatible version.

Thank you for your work.

**Comment by spazzarama** [comment]
@RussKie thanks for the feedback. I've been very busy so haven't progressed this any further yet. Basically the plan is to remove IPC from the core project (except for some very basic internals), and then leave it to the consumers of this library to implement an appropriate IPC solution (less opinionated that way too) - perhaps with some examples.

**Comment by RussKie** [comment]
Totally understand, I've been under pump myself :D

Looking forward to the new version with examples how we could use it.

**Comment by netcorefan1** [comment]
I just saw this issue mentioned a couple of days ago and would like to ask if there are some news. I abandoned .Net Framework long time ago and now I am trying to keep everything in .Net 5, but I am having troubles with EH. The Nuget package does not work and IPC is not supported. I am not sure how I can connect my projects to EH.

There is some sort of fork called [CoreHook](https://github.com/unknownv2/CoreHook) which is very nice, but I am not sure if the project has been [abandoned](https://github.com/unknownv2/CoreHook/issues/135) and I don't feel to risk.
Hope to see soon the new EH up and running. Not only is a wonderful project, but it's something we all need.

**Comment by spazzarama** [comment]
@netcorefan1 yes I still plan to do a .Net standard compatible version. Still a problem with available time for now however.

**Comment by RussKie** [comment]
@spazzarama do you have any drafts to peak at?

**Comment by spazzarama** [comment]
Not yet, but basically everything will work except IPC. My plan is to remove embedded IPC, and instead users can choose from the variety of other awesome IPC solutions out there.

**Comment by netcorefan1** [comment]
I am glad to see that you are going to remove any embedded IPC and leave the implementation to the final user. This was the first thing I would have gone to suggest through a new issue request if I was not aware of this.
The best would be that we will see the new version directly replacing the current Nuget package.
For pre-releases, please let us know in this thread, in this way we can help you.
Hope to hear soon good news from you because my project is stalled and when my projects remain stalled for too long time, most of the times they never get completed.

**Comment by spazzarama** [comment]
An initial commit of the EasyHook library that builds under .NET Standard is on this branch: https://github.com/EasyHook/EasyHook/tree/feature/netstandard

 1. It currently has no support for remoke hooks (injection), only LocalHooks.
 2. COMClassInfo cannot determine method indexes (must manually provide method indexes)
 3. No nuget / build scripts etc
 4. EasyHook.Tests passed, however you need to manually copy all dependencies for the moment

For now, just manually build EasyHook / EasyHookDll and copy to your .NET Core solution to use LocalHooks

**Comment by netcorefan1** [comment]
Uahhh !! Great and unexpected news! Unfortunately my projects only use remote hooks, therefore I can't do anything. I already managed to implement a rude IPC replacement and as soon as you implement remote hooking support I will let you know how it goes. Anyway, it's great to know that things started to move on!

**Comment by yegor-pelykh** [comment]
Hi @justinstenning,
сan we expect a .NET / .NET Core version to be released soon?

**Comment by justinstenning** [comment]
@yegor-pelykh no, I haven’t had time to look at this since the pre-release supporting localhooks only.
